### ISSUE TITLE
Task "migrate-syntax" fails due to colons within path strings on Windows

### ISSUE THREAD

**Issue by aholland** [scalacenter/scala3-migrate#176]
Task **`migrate-syntax`** falls over while trying to normalize Windows paths, apparently by including colons (":") in the path string being normalized.

```
[IJ]migrate-syntax aholland-scala-utils-scala_3
[success] Total time: 0 s, completed 24 Mar 2021, 12:16:11
[success] Total time: 2 s, completed 24 Mar 2021, 12:16:13
[success] Total time: 2 s, completed 24 Mar 2021, 12:16:15
[info] We are going to fix some syntax incompatibilities on aholland-scala-utils-scala_3
[error]
[error]
[error] Failed fixing the syntax for aholland-scala-utils-scala_3 project
[error] Illegal char <:> at index 88: /home/runner/work/scala3-migrate/scala3-migrate/scalafix/rules/target/scala-2.13/classes:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.13.5/scala-library-2.13.5.jar:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/ch/epfl/scala/scalafix-core_2.13/0.9.26/scalafix-core_2.13-0.9.26.jar:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/ch/epfl/scala/scalafix-rules_2.13/0.9.26/scalafix-rules_2.13-0.9.26.jar:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scalameta/scalameta_2.13/4.4.10/scalameta_2.13-4.4.10.jar:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/com/googlecode/java-diff-utils/diffutils/1.3.0/diffutils-1.3.0.jar:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/com/geirsson/metaconfig-typesafe-config_2.13/0.9.10/metaconfig-typesafe-config_2.13-0.9.10.jar:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/modules/scala-collection-compat_2.13/2.4.1/scala-collection-compat_2.13-2.4.1.jar:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scalameta/semanticdb-scalac-core_2.13.5/4.4.10/semanticdb-scalac-core_2.13.5-4.4.10.jar:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scalameta/parsers_2.13/4.4.10/parsers_2.13-4.4.10.jar:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scalap/2.13.5/scalap-2.13.5.jar:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/com/geirsson/metaconfig-core_2.13/0.9.10/metaconfig-core_2.13-0.9.10.jar:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/com/typesafe/config/1.2.1/config-1.2.1.jar:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-compiler/2.13.5/scala-compiler-2.13.5.jar:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scalameta/trees_2.13/4.4.10/trees_2.13-4.4.10.jar:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/org/
typelevel/paiges-core_2.13/0.3.0/paiges-core_2.13-0.3.0.jar:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/com/lihaoyi/pprint_2.13/0.5.9/pprint_2.13-0.5.9.jar:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-reflect/2.13.5/scala-reflect-2.13.5.jar:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/org/jline/jline/3.19.0/jline-3.19.0.jar:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/net/java/dev/jna/jna/5.3.1/jna-5.3.1.jar:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scalameta/common_2.13/4.4.10/common_2.13-4.4.10.jar:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/com/thesamet/scalapb/scalapb-runtime_2.13/0.10.11/scalapb-runtime_2.13-0.10.11.jar:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scalameta/fastparse_2.13/1.0.1/fastparse_2.13-1.0.1.jar:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/com/lihaoyi/fansi_2.13/0.2.9/fansi_2.13-0.2.9.jar:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/com/lihaoyi/sourcecode_2.13/0.2.3/sourcecode_2.13-0.2.3.jar:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/com/thesamet/scalapb/lenses_2.13/0.10.11/lenses_2.13-0.10.11.jar:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/com/google/protobuf/protobuf-java/3.11.4/protobuf-java-3.11.4.jar:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/com/lihaoyi/fastparse_2.13/2.3.0/fastparse_2.13-2.3.0.jar:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scalameta/fastparse-utils_2.13/1.0.1/fastparse-utils_2.13-1.0.1.jar:/home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/com/lihaoyi/geny_2.13/0.6.0/geny_2.13-0.6.0.jar
[error] java.base/sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
[error] java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
[error] java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
[error] java.base/sun.nio.fs.WindowsPath.parse(WindowsPath.java:92)
[error] java.base/sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:229)
[error] java.base/java.nio.file.Path.of(Path.java:147)
[error] java.base/java.nio.file.Paths.get(Paths.java:69)
[error] migrate.internal.AbsolutePath.toNio(AbsolutePath.scala:12)
[error] migrate.internal.AbsolutePath.<init>(AbsolutePath.scala:10)
[error] migrate.internal.AbsolutePath$.apply(AbsolutePath.scala:9)
[error] migrate.internal.AbsolutePath$.$anonfun$from$1(AbsolutePath.scala:25)
[error] scala.util.Try$.apply(Try.scala:210)
[error] migrate.internal.AbsolutePath$.from(AbsolutePath.scala:25)
[error] migrate.internal.Classpath$.$anonfun$from$1(Classpath.scala:24)
[error] scala.collection.immutable.List.map(List.scala:246)
[error] migrate.internal.Classpath$.from(Classpath.scala:24)
[error] migrate.utils.ScalafixService$.internalRules$lzycompute(ScalafixService.scala:70)
[error] migrate.utils.ScalafixService$.internalRules(ScalafixService.scala:70)
[error] migrate.utils.ScalafixService$.$anonfun$from$1(ScalafixService.scala:86)
[error] scala.util.Success.flatMap(Try.scala:258)
[error] migrate.utils.ScalafixService$.from(ScalafixService.scala:85)
[error] migrate.interfaces.MigrateImpl.$anonfun$migrateSyntax$6(MigrateImpl.scala:70)
[error] scala.util.Success.flatMap(Try.scala:258)
[error] migrate.interfaces.MigrateImpl.$anonfun$migrateSyntax$3(MigrateImpl.scala:67)
[error] scala.util.Success.flatMap(Try.scala:258)
[error] migrate.interfaces.MigrateImpl.$anonfun$migrateSyntax$2(MigrateImpl.scala:66)
[error] scala.util.Success.flatMap(Try.scala:258)
[error] migrate.interfaces.MigrateImpl.migrateSyntax(MigrateImpl.scala:65)
[error] migrate.ScalaMigratePlugin$.$anonfun$migrateSyntaxImpl$3(ScalaMigratePlugin.scala:223)
[error] scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error] scala.util.Try$.apply(Try.scala:213)
[error] migrate.ScalaMigratePlugin$.$anonfun$migrateSyntaxImpl$1(ScalaMigratePlugin.scala:219)
[error] migrate.ScalaMigratePlugin$.$anonfun$migrateSyntaxImpl$1$adapted(ScalaMigratePlugin.scala:207)
[error] scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error] sbt.std.Transform$$anon$4.work(Transform.scala:68)
[error] sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[error] sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[error] sbt.Execute.work(Execute.scala:291)
[error] sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[error] sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error] sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[error] java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[error] java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
[error] java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
[error] java.base/java.lang.Thread.run(Thread.java:832)
```

**Comment by aholland** [comment]
Just realised this had already been reported in https://github.com/scalacenter/scala3-migrate/issues/150

**Comment by mlachkar** [comment]
Thank you for this report. It's super useful with the stack trace.

**Comment by mlachkar** [comment]
This issue is now fixed and released under `0.3.1`. Thanks again for this report.
